src/shared/install.c

```diff
@@ -2423,33 +2431,66 @@ int unit_file_query_preset(UnitFileScope scope, const char *root_dir, const char
                         if (strchr(COMMENTS, *l))
                                 continue;
 
-                        parameter = first_word(l, "enable");
-                        if (parameter) {
-                                if (fnmatch(parameter, name, FNM_NOESCAPE) == 0) {
-                                        log_debug("Preset file says enable %s.", name);
-                                        return 1;
-                                }
+                        if ((parameter = first_word(l, "enable"))) {
+                                name = strdupa(parameter);
+                                state = new0(PresetState, 1);
```

Comment:
No need to allocate space for this, it's just an int right?  the pointer can represent the state, just use the INT_TO_PTR() macro from `basic/macro.h`.